python_modules/dagster/dagster/_utils/caching_instance_queryer.py

```diff
@@ -54,6 +56,8 @@ def __init__(self, instance: DagsterInstance):
 
         self._dynamic_partitions_cache: Dict[str, Sequence[str]] = {}
 
+        self._evaluation_time = evaluation_time
```

Comment:
alternatively, we could just make this a required argument

Reply:
that sounds good to me. Originally I didn't set it because there's a couple other places the caching instance queryer is used without eval time, but I think we can safely assume in these callsites that the partitions in each partitions def shouldn't change